test/jdk/javax/swing/plaf/nimbus/TestNimbusBGColor.java

```diff
@@ -22,29 +22,76 @@
  */
 /*
  * @test
- * @bug 6789980
+ * @bug 8058704
  * @key headful
- * @summary  Verifies if Nimbus honor JEditorPane background color
- * @run main TestNimbusJEditorPaneBgColor
+ * @summary  Verifies if Nimbus honor JTextPane background color
+ * @run main TestNimbusBGColor
  */
 import java.awt.Color;
 import java.awt.Dimension;
 import java.awt.Point;
 import java.awt.Rectangle;
 import java.awt.Robot;
 import javax.swing.JFrame;
+import javax.swing.JTextPane;
 import javax.swing.JEditorPane;
 import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
 
-public class TestNimbusJEditorPaneBgColor {
+public class TestNimbusBGColor {
 
     static JFrame frame;
     static volatile Point pt;
     static volatile Rectangle bounds;
+    static Robot robot;
 
     public static void main(String[] args) throws Exception {
-        Robot robot = new Robot();
+        robot = new Robot();
+        testTextPane();
+        testEditorPane();
+    }
+
+    private static void testTextPane() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                try {
+                    UIManager.setLookAndFeel("javax.swing.plaf.nimbus.NimbusLookAndFeel");
+                } catch (Exception checkedExceptionsPleaseDie) {
+                    throw new RuntimeException(checkedExceptionsPleaseDie);
+                }
+                JTextPane tp = new JTextPane();
+                tp.setForeground(Color.WHITE);
+                tp.setBackground(Color.BLACK);
+                tp.setText("This text should be white on black");
```

Comment:
I meant to pass a component as a parameter to a generic test method.

```java
private static void testComponent(ComponentCreator<? extends JTextComponent> creator) {
        // Set Look and Feel
        JComponent tc = creator.createComponent();
        // set foreground etc and verify
}
```

I implemented [what I had in mind](https://github.com/aivanov-jdk/jdk/commit/ba9cbc12c10ba6950a3cd491a9c9c022ab0dad2f). I don't think it's important what background color you use for each component. The updated test fails without the fix and passes with your fix.